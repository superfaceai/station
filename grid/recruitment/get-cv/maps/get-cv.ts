/* eslint-disable jest/no-export */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';

import { buildSuperfaceTest } from '../../../test-config';

export const getCVTest = (
  provider: string,
  testInputs: {
    validCandidateId: string;
    invalidCandidateIds: string[];
  },
  options?: RecordingProcessOptions
): void => {
  describe(`recruitment/get-cv/${provider}`, () => {
    let superface: SuperfaceTest;

    describe('GetCV', () => {
      beforeAll(() => {
        superface = buildSuperfaceTest({
          profile: 'recruitment/get-cv',
          provider,
          useCase: 'GetCV',
        });
      });

      describe('when geting CV with valid candidate ID', () => {
        it('returns list of candidates', async () => {
          const result = await superface.run(
            {
              input: {
                candidateId: testInputs.validCandidateId,
              },
              testName: 'get CV with valid candidate ID',
            },
            options
          );

          expect(() => result.unwrap()).not.toThrow();
          expect(result).toMatchSnapshot();
        });
      });

      testInputs.invalidCandidateIds.forEach(invalidCandidateId => {
        describe(`when geting CV with invalid candidate ID ${invalidCandidateId}`, () => {
          it('returns candidate not found error', async () => {
            const result = await superface.run(
              {
                input: {
                  candidateId: invalidCandidateId,
                },
                testName: `get CV with invalid candidate ID ${invalidCandidateId}`,
              },
              options
            );

            expect(() => result.unwrap()).toThrow();
            expect(result).toMatchSnapshot();
          });
        });
      });
    });
  });
};

export const listCandidatesBreezyHRSpecificTest = (
  options?: RecordingProcessOptions
): void => {
  const provider = 'breezy-hr';

  describe(`recruitment/get-cv/${provider}`, () => {
    let superface: SuperfaceTest;

    describe('GetCV', () => {
      beforeAll(() => {
        superface = buildSuperfaceTest({
          profile: 'recruitment/get-cv',
          provider,
          useCase: 'GetCV',
        });
      });

      describe('when specified company does not exist', () => {
        let companyId: string | undefined;

        beforeAll(() => {
          companyId = process.env.BREEZY_HR_COMPANY_ID;

          process.env.BREEZY_HR_COMPANY_ID = '1b111c1111ef11';
        });

        afterAll(() => {
          process.env.BREEZY_HR_COMPANY_ID = companyId;
        });

        it('returns error', async () => {
          const result = await superface.run(
            {
              input: {
                jobId: 'JOB_ID',
              },
            },
            options
          );

          expect(() => result.unwrap()).toThrow();
          expect(result).toMatchSnapshot();
        });
      });
    });
  });
};

