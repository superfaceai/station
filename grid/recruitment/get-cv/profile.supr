"""
Get CV
Gets candidate CV document.
"""
name = "recruitment/get-cv"
version = "1.0.0"

"""
Get CV
Get CV document of the candidate by candidate ID. Returns either URL or binary data of the CV document.
"""
usecase GetCV safe {
  input {
    """
    Contact ID
    Uniquie identifier of the candidate. Use recruitment/list-candidates profile to list candidate IDs.
    """
    candidateId! string!
  }

  result {
    """
    Created At
    ISO8601 timestamp when was the CV uploaded.
    """
    createdAt string!
    """
    File Name
    The name of CV document.
    """
    fileName string!
    """
    File Content
    The content of the CV document.
    """
    fileContent
    """
    File URL
    The URL of the CV document.
    """
    fileUrl string!
  }

  error Error

  example Success {
    input {
      candidateId = "CANDIATE_ID"
    }
    result {
      createdAt = "2023-01-05T10:04:58.325Z"
      fileName = "cv.pdf"
      fileContent = "<BinaryData>"
    }
  }

  example Fail {
    input {
      candidateId = "WRONG_CANDIDATE_ID"
    }
    error {
      title = "Not Found"
      detail = "Candidate ID \"WRONG_CANDIDATE_ID\" do not exist."
      code = "CandidateNotFound"
    }
  }
}

model Error {
  """
  Error Title
  A short, human-readable summary of the problem type.
  """
  title!
  """
  Error Detail
  A human-readable explanation specific to this occurrence of the problem.
  """
  detail
  """
  Code
  Error code.
  """
  code! ErrorCode!
  """
  Rate limit
  Rate limit details.
  """
  rateLimit RateLimit
}

model RateLimit {
  """
  Bucket
  Different parts of API may have different rate limits.
  Bucket identifies to which part of API the rate limits apply.
  """
  bucket string
  """
  Total requests
  Total requests available in the time window.
  """
  totalRequests number
  """
  Remaining requests
  Remaining requests available in the time window.
  """
  remainingRequests number
  """
  Remaining requests available in percents
  Remaining requests available in the time window in percents.
  """
  remainingRequestsPercentage number
  """
  Reset timestamp
  Timestamp when the rate limits will reset (in  Unix Timestamp format). Available only for providers without rolling rate limit window.
  """
  resetTimestamp number
}

model ErrorCode enum {
  """
  Bad request
  The request was invalid or cannot be otherwise served. 
  """
  BadRequest
  """
  Candidate not found
  The candidate ID does not exist.
  """
  CandidateNotFound
  """
  CV not found
  The CV document has not been found.
  """
  CVDocumentNotFound
  """
  Wrong integration parameter
  The integration parameter value has not been accepted.
  """
  WrongIntegrationParameter
  """
  Rate limit reached
  Returned when a request cannot be served due to the user or app reached max requests quota.
  """
  RateLimitReached
  """
  Unauthenticated
  There was a problem authenticating your request. Check that you are passing valid credentials.
  """
  Unauthenticated
  """
  Unauthorized
  The request is understood, but it has been refused or access is not allowed.
  """
  Unauthorized
  """
  Unknown error
  Returned when an unexpected error occured.
  """
  UnknownError
}
