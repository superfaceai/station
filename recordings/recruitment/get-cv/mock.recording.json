{
  "recruitment/get-cv/mock/GetCV": {
    "f9dc26dfb5ba0309eac14236c2781671": [],
    "eb485296977fbd9863bb8ef67c5e043a": []
  }
}