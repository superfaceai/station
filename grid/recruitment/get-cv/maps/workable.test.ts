import { getCVTest, getCVWorkableSpecificTest } from './get-cv';

getCVTest('workable', {
  validCandidateId: 'd686952',
  invalidCandidateIds: ['INVALID_CANDIDATE_ID'],
});

getCVWorkableSpecificTest();
