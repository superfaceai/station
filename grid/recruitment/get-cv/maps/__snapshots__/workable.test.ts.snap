// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recruitment/get-cv/workable GetCV when geting CV with invalid candidate ID INVALID_CANDIDATE_ID returns candidate not found error 1`] = `
Err {
  "error": "JessieError: Error in Jessie script
TypeError: Cannot read properties of undefined (reading 'fileUrl')
AST Path: definitions[0].statements[2].value.fields[0].value
Original Map Location: Line 23, column 10",
}
`;

exports[`recruitment/get-cv/workable GetCV when geting CV with valid candidate ID returns CV document 1`] = `
Ok {
  "value": Object {
    "cv": Object {
      "documentUrl": "https://workablehr.s3.amazonaws.com/uploads/587179/2887832/224945280/download/cv-sample.pdf?X-Amz-Expires=60000&X-Amz-Date=20230109T094538Z&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIR2MIE33PRKXR4CA%2F20230109%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-SignedHeaders=host&X-Amz-Signature=50f85c7c719a758570f4b13d3c2eb18c3bf7177d2c3a10ddae55725d7d03a94b",
    },
    "rateLimit": Object {
      "bucket": "workable",
      "remainingRequests": 8,
      "remainingRequestsPercentage": 80,
      "resetTimestamp": 1673257548,
      "totalRequests": 10,
    },
  },
}
`;

exports[`recruitment/get-cv/workable GetCV when specified subdomain does not exist returns error 1`] = `
Err {
  "error": "MappedError: Expected error
Properties: {
  \\"title\\": \\"Wrong Integration Parameter\\",
  \\"detail\\": \\"Subdomain  invalid-subdomain do not exist.\\",
  \\"code\\": \\"WrongIntegrationParameter\\",
  \\"rateLimit\\": {
    \\"bucket\\": \\"workable\\"
  }
}
AST Path: definitions[0].statements[0].statements[0]
Original Map Location: Line 9, column 5",
}
`;
